lib/motors/PID.cpp

```diff
@@ -0,0 +1,56 @@
+#include "PID.hpp"
+
+void PIDController::PIDController_Init()
+{
+  /* Clear controller variables */
+  integrator      = 0.0f;
+  prevError       = 0.0f;
+  differentiator  = 0.0f;
+  prevMeasurement = 0.0f;
+
+  out = 0.0f;
+};
+
+float PIDController::PIDController_Update(float setpoint, float measurement)
+{
+  // Error signal
+  float error = setpoint - measurement;
+
+  // Proportional term
+  float proportional = Kp * error;
+
+  // Integral term
+  integrator = integrator + 0.5f * Ki * T * (error + prevError);
+
+  // Integrator Anti wind-up (dynamic integrator clamping)
```

Comment:
real nitpicky, but capitalisation is a bit random through a lot of comments. 

btw, this is a good use of comments. generally, we want comments to explain ***why***, whereas the code should be self explanatory as to the ***how*** in most cases (if you do something really weird that is necessary that is worth commenting tho)